core/src/rigidbody.cpp

```diff
@@ -0,0 +1,289 @@
+#include <knoting/rigidbody.h>
+
+namespace knot {
+namespace components {
+RigidBody::RigidBody() : m_physics(NULL), m_dynamic(NULL), m_static(NULL), m_scene(NULL), m_material(NULL) {}
+
+RigidBody::~RigidBody() {
+    if (m_dynamic) {
+        m_dynamic->release();
+        m_dynamic = NULL;
+    } else {
+        m_static->release();
+        m_static = NULL;
+    }
+    m_material->release();
+    m_material = NULL;
+    m_scene = NULL;
+    m_physics = NULL;
+}
+
+void RigidBody::on_awake() {}
+void RigidBody::on_destroy() {
+    if (m_dynamic) {
+        m_scene->removeActor(*m_dynamic);
+    } else if (m_static) {
+        m_scene->removeActor(*m_static);
+    }
+}
+
+vec3 RigidBody::get_position() const {
+    vec3 v;
+    if (m_dynamic) {
+        v = vec3(m_dynamic->getGlobalPose().p.x, m_dynamic->getGlobalPose().p.y, m_dynamic->getGlobalPose().p.z);
+
+    } else {
+        v = vec3(m_static->getGlobalPose().p.x, m_static->getGlobalPose().p.y, m_static->getGlobalPose().p.z);
+    }
+    return v;
+}
+
+quat RigidBody::get_rotation() const {
+    quat q;
+    if (m_dynamic) {
+        q = quat(m_dynamic->getGlobalPose().q.w, m_dynamic->getGlobalPose().q.x, m_dynamic->getGlobalPose().q.y,
+                 m_dynamic->getGlobalPose().q.z);
+    } else {
+        q = quat(m_static->getGlobalPose().q.w, m_static->getGlobalPose().q.x, m_static->getGlobalPose().q.y,
+                 m_static->getGlobalPose().q.z);
+    }
+    return q;
+}
+
+physx::PxMaterial* RigidBody::get_pxmaterial() {
+    return m_material;
+}
+
+void RigidBody::set_transform(const vec3& position, const quat& rotation) {
+    if (m_dynamic == NULL && m_static == NULL) {
+        return;
+    }
+    float p_x = position.x;
+    float p_y = position.y;
+    float p_z = position.z;
+    float r_w = rotation.w;
+    float r_x = rotation.x;
+    float r_y = rotation.y;
+    float r_z = rotation.z;
```

Comment:
It doesn't appear that a re-declaration of every variable is needed, pass these variables directly.